### ISSUE TITLE
rollup issue with undeclared variable `process`

### ISSUE THREAD

**Issue by whtevn** [driftyco/ionic-app-scripts#130]
#### Short description of the problem:
When I import the aws-sdk, rollup complains that there is an illegal assignment of the variable `process`. When I use rollup independently with the aws-sdk, there is no issue.

#### What behavior are you expecting?

I am expecting the information in the AWS object that is imported to be logged out, and for the build to succeed

**Steps to reproduce:**
1. `npm install aws-sdk` 
2. `import AWS` from `aws-sdk` in any `.ts` file within a new ionic v2 project
3. `ionic build` and the error message will be revealed to you

**Which @ionic/app-scripts version are you using?**

@ionic/app-scripts version 0.0.30

**Other information:** (e.g. stacktraces, related issues, suggestions how to fix, stackoverflow links, forum links, etc)

the code I am using to produce this error is

```
import AWS from 'aws-sdk';
console.log(AWS);
```

and that gives the following build error

```
Error: Illegal reassignment to import 'process'
    at error (/Users/evan/Documents/dev/dn/flashtastic-ionic/node_modules/rollup/dist/rollup.js:5404:12)
    at disallowIllegalReassignment (/Users/evan/Documents/dev/dn/flashtastic-ionic/node_modules/rollup/dist/rollup.js:5429:4)
    at AssignmentExpression.bind (/Users/evan/Documents/dev/dn/flashtastic-ionic/node_modules/rollup/dist/rollup.js:5492:3)
    at /Users/evan/Documents/dev/dn/flashtastic-ionic/node_modules/rollup/dist/rollup.js:5149:50
    at ExpressionStatement.eachChild (/Users/evan/Documents/dev/dn/flashtastic-ionic/node_modules/rollup/dist/rollup.js:5166:5)
    at ExpressionStatement.bind (/Users/evan/Documents/dev/dn/flashtastic-ionic/node_modules/rollup/dist/rollup.js:5149:7)
    at BlockStatement.bind (/Users/evan/Documents/dev/dn/flashtastic-ionic/node_modules/rollup/dist/rollup.js:5615:9)
    at /Users/evan/Documents/dev/dn/flashtastic-ionic/node_modules/rollup/dist/rollup.js:5149:50
    at IfStatement.eachChild (/Users/evan/Documents/dev/dn/flashtastic-ionic/node_modules/rollup/dist/rollup.js:5166:5)
    at IfStatement.bind (/Users/evan/Documents/dev/dn/flashtastic-ionic/node_modules/rollup/dist/rollup.js:5149:7)
```

I have tracked this down to a particular block in `aws-sdk/lib/browser_loader.js`, which if removed alleviates the build error (but causes an application error)

```
if (typeof process === 'undefined') {
  process = {
    browser: true
  };
}
```

it is also worth noting that the `process` variable is not declared with `var` anywhere in that file. If I add the `var` declaration, the build error goes away, but the application error returns.

**Comment by whtevn** [comment]
forgot to mention, aws-sdk version is 2.6.7

**Comment by danbucholtz** [comment]
@whtevn,

We won't be able to rollup this library at this time. I have created an issue for you on the `aws-sdk` repo.

https://github.com/aws/aws-sdk-js/issues/1175

For now, the bet best will be to download the .js file from [here](https://github.com/aws/aws-sdk-js#in-the-browser) and store it somewhere in the `src/assets` directory. I would then create an entry in the `index.html` file for the `<script>` tag.

Everything will be copied to `www` and should just work.

Check out this [doc](http://ionicframework.com/docs/v2/resources/app-scripts/) (at the very bottom, sorry, we're improving these as we speak) with how to use a lib via a script tag.  Basically, throw this at the top of where you want to use the library:

```
declare const jQuery:any;
```

Substitute `jQuery` with the variable off of `window` that you'd use to access the lib.

Sorry about this. I wish it would work with Rollup. Please let me know if you have any questions.

Thanks,
Dan

**Comment by danbucholtz** [comment]
Ahh! Actually, I was able to get it working by overriding the `rollup` config and removing `globals` from the plugins as seen here.

https://github.com/driftyco/ionic-app-scripts/blob/master/config/rollup.config.js#L48

May be worth a shot.

Thanks,
Dan

**Comment by whtevn** [comment]
First of all, thanks so much for your help so far. I was able to remove the `globals()` call in that config file I copied from the file you linked, and that seemed to have the same effect as removing the block of code or adding the variable. The build error goes away, but an application error comes in its place. 

```
main.js:62101Uncaught TypeError: Cannot read property 'memoizedProperty' of undefined
```

which comes from...

```
import AWS from 'aws-sdk';
console.log(AWS);     // <------- a result of this line in particular
```

which I am currently looking into, but so far not having a lot of luck. I am new to TS, so there is a fine chance I'm missing something obvious.

I tried to confirm the issue by putting `globals()` back in place in the rollup config, removing the `aws-sdk` npm package, and addressing a locally saved copy of the SDK in a `<script>` tag, but I'm apparently not getting something. 

```
declare const AWS:any;

console.log(window.AWS);  // BUILD FAILS - ngc: Error: Error at /Users/evan/Documents/dev/dn/flashtastic-ionic/.tmp/app/app.component.ts:8:20: Property 'AWS' does not exist on type 'Window'.

console.log(AWS);                // APP FAILS - Uncaught ReferenceError: AWS is not defined
```

sorry for the additional questions, but I'm really hoping that I can get this working. I am looking into the `memoizedProperty` TypeError mentioned above. If you have any advice or direction, I would really appreciate it. Thanks again for taking the time to do this.
